keystore/keyloader/kms/kms_per_client_key_mapper.go

```diff
@@ -0,0 +1,49 @@
+package kms
+
+import (
+	"errors"
+
+	"github.com/cossacklabs/acra/keystore"
+)
+
+// KmsEncryptor errors
+var (
+	ErrMissingKeyPurpose     = errors.New("key purpose is required for keyID creating")
+	ErrUnsupportedKeyPurpose = errors.New("unsupported KeyPurpose option provided")
+	ErrEmptyClientIDProvided = errors.New("empty clientID in key context")
+	ErrEmptyZoneIDProvided   = errors.New("empty zoneID in key context")
+)
+
+// KeyMapper Implement KeyMapper interface for `kms_per_client` strategy
+type KeyMapper struct{}
+
+// NewKMSPerClientKeyMapper create new KeyMapper
+func NewKMSPerClientKeyMapper() *KeyMapper {
+	return &KeyMapper{}
+}
+
+// GetKeyID implementation method of KeyMapper interface
+func (k *KeyMapper) GetKeyID(ctx keystore.KeyContext) ([]byte, error) {
+	if ctx.Purpose == "" {
```

Comment:
can we add unit-tests for this new code to check all expected errors in expected cases? to freeze current state in tests and simplify next changes without risks of breaking something)

Reply:
Sure!